test/BUILD.gn

```diff
@@ -557,6 +557,14 @@ if (!is_android) {
       "//brave/browser/extensions/brave_extension_functional_test.h",
       "//brave/browser/extensions/brave_extension_provider_browsertest.cc",
       "//brave/browser/extensions/brave_theme_event_router_browsertest.cc",
+      "//brave/browser/farbling/brave_enumeratedevices_farbling_browsertest.cc",
```

Comment:
This is definitely not how we want to fix these issues for tests, we want to remove files from test/BUILD.gn, not add them. Why was `check_includes` false in the first place for the original target?

Reply:
> This is definitely not how we want to fix these issues for tests, we want to remove files from test/BUILD.gn, not add them.

My bad, I thought a similar approach to what we've been doing for production-related files would be followed for tests but even like that I think I misunderstood the implementation because what I'm doing here is to move things into //brave/tests, not even //chrome/tests (which would be more like the equivalent of what we're doing with //chrome/browser).

In any case, I'll close this PR for now as it seems now to me that for tests is probably better to split things into mutliple & smaller targets, set the dependencies right for each of them and get rid of the //brave/tests::brave_browser_tests target. 

@bridiver Does that sound right?

> Why was `check_includes` false in the first place for the original target?

`check_includes` was `false` because otherwisea few instances of the following error would happen:
```
ERROR at //brave/browser/farbling/brave_navigator_devicememory_farbling_browsertest.cc:12:11: Include not allowed.
#include "brave/browser/extensions/brave_base_local_data_files_browsertest.h"
          ^-----------------------------------------------------------------
It is not in any dependency of
  //brave/browser/farbling:browser_tests
The include file is in the target(s):
  //brave/test:brave_browser_tests
which should somehow be reachable.
```